File: 35 - Search Insert Position/binarySearch.js
var searchInsert = function (nums, target) {
    let l = 0
    let r = nums.length - 1
    while (l <= r) {
        let m = parseInt((l + r) / 2)

        if (nums[m] < target) {
            l = m + 1
        } else if (nums[m] > target) {
            r = m - 1
        } else if (nums[m] == target) {
            return m
        }
    }

    if(l > r) {
        return l
    }
    return 0
};

File: 35 - Search Insert Position/binarySearch.go
func searchInsert(nums []int, target int) int {
    l, r := 0, len(nums) -1
    response :=0

    for l <= r {
        m := int((l+r)/2)
        if target > nums[m] {
            l = m +1
        } else if target < nums[m] {
            r = m -1
        } else if nums[m] == target {
            return m
        }
    }

    if l > r {
        return l
    }

    return response
}